src/test/ui/resolve/suggest-type.rs

```diff
@@ -0,0 +1,19 @@
+use std::ffi::CString;
+
+mod foo {
+    fn bar() {}
+}
+
+fn main() {
+    let _ = Cstring::new("hello").unwrap();
+    //~^ ERROR failed to resolve: use of undeclared type or module `Cstring`
+
+    let _ = CStRiNg::new("hello").unwrap();
+    //~^ ERROR failed to resolve: use of undeclared type or module `CStRiNg`
+
+    let _ = cStrIng::new("hello").unwrap();
```

Comment:
`CStRiNg` and `cStrIng` are testing Levenshtein distance implementation, which is already tested in many places.
Instead, I'd rather see test cases for
- the typo being inside a module and the suggested name being inside the same module (this would catch the `ParentScope::module(self.graph_root)` mistake).
- the suggested name being a generic type parameter (this would catch the `early_lookup_typo_candidate` mistake).